components/raftstore-v2/src/operation/query/capture.rs

```diff
@@ -52,14 +52,24 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: raftstore::store::Transport>
     pub fn on_capture_change(&mut self, capture_change: CaptureChange) {
         fail_point!("raft_on_capture_change");
 
-        let apply_router = self.fsm.peer().apply_scheduler().unwrap().clone();
+        let apply_scheduler = self.fsm.peer().apply_scheduler().cloned();
+        let id = self.fsm.peer().region_id();
+        let term = self.fsm.peer().term();
         let (ch, _) = QueryResChannel::with_callback(Box::new(move |res| {
             if let QueryResult::Response(resp) = res && resp.get_header().has_error() {
                 // Return error
                 capture_change.snap_cb.report_error(resp.clone());
                 return;
             }
-            apply_router.send(ApplyTask::CaptureApply(capture_change))
+            if let Some(scheduler) = apply_scheduler {
+                scheduler.send(ApplyTask::CaptureApply(capture_change))
+            } else {
+                let mut resp = cmd_resp::err_resp(raftstore::Error::RegionNotFound(id), term);
```

Comment:
Should we add some logs here since it should be very rare to happen.

Reply:
I presume cdc will handle the error? It's that component's responsibility.